osu.Game/Tests/Visual/SkinnableTestScene.cs

```diff
@@ -27,14 +28,19 @@ public abstract class SkinnableTestScene : OsuGridTestScene
         private Skin specialSkin;
         private Skin oldSkin;
 
+        /// <summary>
+        /// Creates the ruleset for adding the ruleset-specific skin transforming component.
+        /// </summary>
+        [NotNull]
+        protected abstract Ruleset CreateRulesetForSkinProvider();
```

Comment:
Nits: Methods should not be able ctors (done quite a few times in this PR).

Reply:
I really don't want to repeat changing the order due to how many files changed for it, where should it reside? at the bottom of the ctor, or same as it was before?

Reply:
My thought process with asking for it to be higher was to treat it in a similar manner to `RequiredTypes` is - as setup/test declaration code and not an actual method. Unless you mean only this abstract one in the base...?

Reply:
Right, for declaring classes this should reside where protected methods usually reside, applied.